salvo/src/lib/api/source.proto

```diff
@@ -0,0 +1,35 @@
+syntax = "proto3";
+
+package salvo;
+
+// Capture the location of sources needed for the benchmark
+message SourceRepository {
+  // Specify whether this source location is Envoy or NightHawk
+  oneof identity {
+    bool envoy = 1;
+    bool nighthawk = 2;
+  }
+
+  // Specify the location of the source repository on disk. If specified
+  // this location is used to determine the origin url, branch, and commit
+  // hash. If not specified, the remaining fields must be populated
+  string location = 3;
+
+  // Specify the remote location of the repository. This is ignored if
+  // the source location is specified.
+  string url = 4;
```

Comment:
In the comments for these fields - can we include one valid example for each of the field values?

Reply:
Updated